cirq-core/cirq/qis/channels.py

```diff
@@ -50,11 +50,9 @@ def kraus_to_choi(kraus_operators: Sequence[np.ndarray]) -> np.ndarray:
         Choi matrix of the channel specified by kraus_operators.
     """
     d = np.prod(kraus_operators[0].shape, dtype=np.int64)
-    c = np.zeros((d, d), dtype=np.complex128)
-    for k in kraus_operators:
-        v = np.reshape(k, d)
-        c += np.outer(v, v.conj())
-    return c
+    k = np.asarray(kraus_operators)
+    flat_ops = k.reshape((-1, d))
```

Comment:
As a matter of principle, I think we should prefer errors to be raised as soon as they occur and not propagate far away from the cause since if that happens the error is harder to diagnose and fix.

For this reason, I'd rather we be explicit about the expected array shape and say `reshape((len(kraus_operators), d))` rather than `reshape((-1, d))`.

---

BTW: This `-1` hides a concept that has a name: it is the Choi rank (assuming we were indeed pasted in a Kraus, i.e. minimal representation)! You can use this to make the code read well and (integrating the other comment) say:

```python
choi_rank = len(kraus_operators)
k = np.reshape(kraus_operators, (choi_rank, d))
return np.einsum('bi,bj->ij',, k, k.conj())
```

This makes it clear that `b` index in the einsum ranges over the Kraus operators.